1-mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,33 @@
-function setAlarm() {}
+// 1. When the `Set Alarm` button is clicked, get the value of the input field
+// 2. When you have the input field value, set the title to the correct value
+// 3. Work out how to update the `Time Remaining` title every second
+// 4. When the remaining time left is 0, play the alarm sound
+
+
+    
+  function setAlarm() {
+  let timer = document.querySelector("#alarmSet").value;
+  setTime(timer);
+  
+  let setter= setInterval(function () {
+    if (timer ===0){
+      clearInterval(setter);
+    playAlarm();
+    } else {
+      timer--;
+      setTime(timer);
+    }
+}, 1000);
+}
+
+function setTime (time){
+  const hour = Math.floor(time / 60);
+  const minute = time % 60;
+  let timeLeft = document.querySelector("#timeRemaining");
+  timeLeft.innerText = `Time Remaining: ${String(hour).padStart(2,"0")}:${String(minute).padStart(2,"0")}`;
+}
+
+
 
```

Comment:
Nice job getting the counter to work, and parsing the time nicely into minutes! I see that the interval also clears when it gets to zero - you should consider what might happen when we hit the button multiple times. What would we like or expect to happen as a user? (Hint: you'll have to clear the interval under different conditions)